bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/quickaccess/providers/CommandProvider.java

```diff
@@ -117,6 +154,33 @@ private void retrieveCommand(final String currentCommandId) {
 		}
 	}
 
+	private boolean filteredWithActivitySupport(ParameterizedCommand pc) {
+		IWorkbenchActivitySupport workbenchActivitySupport = PlatformUI.getWorkbench().getActivitySupport();
+		// fetch qualified name: plugin-is/command-name.
+		String commandId = calculateQualifiedCommandId(pc.getCommand());
+
+		if (commandId == null) {
+			return false;
+		}
+
+		IIdentifier identifier = workbenchActivitySupport.getActivityManager().getIdentifier(commandId);
+		if (!identifier.isEnabled()) {
+			return true;
+		}
+		return false;
+	}
```

Comment:
Move `workbenchActivitySupport ` closer to its usage (it's not needed before that), fix typo in comment and add direct return instead of `if (...) return true`:

```suggestion
	private boolean filteredWithActivitySupport(ParameterizedCommand pc) {
		// fetch qualified name: plugin-id/command-name.
		String commandId = calculateQualifiedCommandId(pc.getCommand());

		if (commandId == null) {
			return false;
		}

		IWorkbenchActivitySupport workbenchActivitySupport = PlatformUI.getWorkbench().getActivitySupport();
		IIdentifier identifier = workbenchActivitySupport.getActivityManager().getIdentifier(commandId);
		return !identifier.isEnabled();
	}
```

Reply:
Fixed.